main.tf

```diff
@@ -30,7 +30,7 @@ resource "cosign_sign" "signature" {
   conflict = "REPLACE"
 }
 
-locals { archs = toset(concat(["index"], data.apko_config.this.config.archs)) }
+locals { archs = toset(concat(["index"], try(jsondecode(var.config).archs, []))) }
```

Comment:
Does this mean that `var.extra_packages` or `var.default_annotations` are derived from resources not known at apply time because this wouldn't work if it were var.config 🤔